### DISCUSSION TITLE
Set layer permission using REST API

### DISCUSSION THREAD

**Discussion by jonsken** [GeoNode/geonode#12773] [Q&A] [answered]
Can I set permission to a layer after or during uploading it using REST API? Using the UI I can adjust the permissions using these dropdowns:
![grafik](https://github.com/user-attachments/assets/e23c33a6-af19-4d60-b505-e7e9741a60ae)

Can I do something similar using REST API, like set `Anyone` and `Registered members` to `None` and also add a specific `Group` for the layer

I tried a few things like:
```
def create_group(title, slug, description):
    payload = {
            "title": title,
            "slug": slug,
            "description": description,
            "access": "private",
        },

    response = requests.put(
        GEONODE_API_URL,
        json=payload,
        auth=(GEONODE_USERNAME, GEONODE_PASSWORD)
    )
    
    if response.status_code == 200:
        print(f"Creating new group successfuly. Status code: {response.status_code}")
        return groups
    else:
        print(f"Failed to create new group. Status code: {response.status_code}")
        return None
```
But `PUT` is not allowed and for `POST` I assume it is not the right structure

**Comment by gannebamm** [comment, marked as answer]
Hi @jonsken,
Here are some examples with curl:

to make RESSOURCE-ID public:

`curl -u admin:PASS --location --request PUT 'https://stable.demo.geonode.org/api/v2/resources/RESSOURCE-ID/permissions' --header 'Content-Type: application/json' --data-raw '{"groups": [],"organizations": [],"users": [{"id": -1,"permissions": "view"}]}'`

To make RESSOURCE-ID private:

`curl -u admin:PASS --location --request PUT 'https://stable.demo.geonode.org/api/v2/resources/RESSOURCE-ID/permissions' --header 'Content-Type: application/json' --data-raw '{"groups": [],"organizations": [],"users": [{"id": -1,"permissions": ""}]}'`

You shall check the `user.id` you would like to grant permissions to.
